File: src/test/java/org/goafabric/personservice/base/extensions/HttpInterceptorTest.java
package org.goafabric.personservice.base.extensions;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.goafabric.personservice.base.extensions.HttpInterceptor;
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import static org.assertj.core.api.Assertions.assertThat;

class HttpInterceptorTest {
    private HttpInterceptor httpInterceptor = new HttpInterceptor();

    @Test
    void preHandle() {
        assertThat(httpInterceptor.preHandle(
                Mockito.mock(HttpServletRequest.class)
                , Mockito.mock(HttpServletResponse.class), new Object()))
                .isTrue();
    }

}

File: src/test/kotlin/org/goafabric/personservice/extensions/HttpInterceptorTest.kt
package org.goafabric.personservice.extensions

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock

internal class HttpInterceptorTest {
    private val httpInterceptor = HttpInterceptor()

    @Test
    fun preHandle() {
        assertThat(
            httpInterceptor.preHandle(
                mock(),
                mock(), Any()
            )
        ).isTrue()
    }

    @Test
    fun getTenantId() {
        assertThat(TenantContext.tenantId).isEqualTo("0")
    }

    @Test
    fun getUserName() {
        assertThat(TenantContext.userName).isEqualTo("anonymous")
    }

}